### DISCUSSION TITLE
Prevent Window Resize on App Start

### DISCUSSION THREAD

**Discussion by wyhinton** [fltk-rs/fltk-rs#777] [Q&A]
When creating a new Window, it seems like the size of the window can sometimes resize automatically. I assume that FLTK is trying to account for some kind of screen screen scaling? Or maybe this is a Win10 thing? But on the monitor that I captured the GIF, the screen scale is 1.0. The scale on my other 4k monitor is 1.25. Is there anyway to have the window appear as the correct size before it appears? 

This can also sometimes cause widgets to resize strangely. 

```rust
use fltk::{
    app, app::*, button::*, enums::*, frame::*, group::*, prelude::*, window::*, widget::*, draw::*,
};
//fltk = { version = "1.0.22"}
fn main(){
    let app = App::default();
    let mut win = Window::new(200, 200, 500, 300, "Pack vs Group Overdraw");
    let mut my_pack= Pack::new(50,50,100,200,"mypack");
    let _button= Button::new(0,0,200,20,"mybutton");
    let _button= Button::new(0,0,200,20,"mybutton");
    let _button= Button::new(0,0,200,20,"mybutton");
    let _button= Button::new(0,0,200,20,"mybutton");
    let mut my_group= Group::new(200,50,100,50,"my group");
    my_group.set_frame(FrameType::FlatBox);
    my_group.set_color(Color::Red);


    let mut frame = Frame::new(200,100,100,100, "Test");
    frame.set_frame(FrameType::FlatBox);
    frame.set_color(Color::Yellow);
    my_group.end();

    let mut frame = Frame::new(250,100,100,100, "Test");
    frame.set_frame(FrameType::FlatBox);
    frame.set_color(Color::Yellow);
    my_pack.end();
    win.end();
    win.show();
    app.run().unwrap();
}
```

![window_resize_issue](https://user-images.githubusercontent.com/38958118/125679921-a1579901-5412-4268-bc61-050bd4ebe1ae.gif)

### Demonstrating the odd resizing effect for my DnD ToDoList example
![window_resize_issue2](https://user-images.githubusercontent.com/38958118/125716097-c53094ea-35d6-4d27-8c8a-0844df4dbf65.gif)

### What the window should look like:
![image](https://user-images.githubusercontent.com/38958118/125716171-2d75e7ff-2ba2-4310-9469-db263453a97c.png)

I'm wondering if this might have something to do with the use of ```Pack```.

**Comment by MoAlyousef** [comment]
I'm not getting that effect on my Windows 10 machine. Is it something recent?
Normally if you're doing the resizing, you can do it before calling win.show(). What happens if you manually set the scale at the beginning of your program `app::set_screen_scale(0, app::screen_scale(0));`?

**Reply by wyhinton** [reply]
No, since before v1.0 I feel like I can remember similar behavior. Kind of reminds me of  [https://github.com/QuantumBadger/Speedy2D/issues/16](https://github.com/QuantumBadger/Speedy2D/issues/16 ). But doubt their is any connection because they are using winit/glutin for their windows.

Sometimes the window appears as expected, but sometimes not. I'll do some more testing to see if I can narrow down the cause.

**Reply by wyhinton** [reply]
> I'm not getting that effect on my Windows 10 machine. Is it something recent?
> Normally if you're doing the resizing, you can do it before calling win.show(). What happens if you manually set the scale at the beginning of your program app::set_screen_scale(0, app::screen_scale(0));?

No effect with ``` app::set_screen_scale(0, app::screen_scale(0))```.

I'm using the same monitor setup as when we were working through the gidplus issue. (launching the app on the active screen in VSCode, then having the app appear on the "non active" monitor).

When I launch the app, then move my mouse over to the non active screen before the window shows, the window draws normally. If I don't, then we get the odd window resizing. 

I can test some different fltk-rs versions to see if this behavior persists, if that'd be helpful.
